File: 1-guia/codigos-1/ej-3/codigo1-2.cpp
int leftiano(int arr[], const std::size_t size) {
    if (size < 2) return 1;
    const std::size_t medio = size / 2;

    int izq[medio], der[medio], sumaIzq{0}, sumaDer{0};

    for (std::size_t i{0}; i < medio; i++) {
        izq[i] = arr[i];
        der[i]   = arr[medio + i];

        sumaIzq += izq[i];
        sumaDer   += der[i];
    }
    return (sumaIzq > sumaDer) 
           ? (leftiano(izq, medio) && leftiano(der, medio))
           : 0;
}

File: 1-guia/codigos-1/ej-1/codigo1-2.py
def merge(izq, der):
    mergeados = []
    i = j = 0

    while i < len(izq) and j < len(der):
        if izq[i] < der[j]:
            mergeados.append(izq[i])
            i += 1
        else:
            mergeados.append(der[j])
            j += 1

    mergeados.extend(izq[i:])
    mergeados.extend(der[j:])
    return mergeados
